packer/build_box.sh

```diff
@@ -0,0 +1,191 @@
+#!/bin/bash -e
+
+#########################################
+#
+#  Global variables
+#
+#########################################
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+#########################################
+#
+#  Usage
+#
+#########################################
+Usage(){
+    echo "Function: this script is used to build vagrant box"
+    echo "usage: $0 [arguments]"
+    echo "    mandatory arguments:"
+    echo "      --OS_VER: The version of "
+    echo "      --RACKHD_VERSION: The version of rackhd debian package"
+    echo "      --RACKHD_APT_REPOSITORY: The apt repository of rackhd debian package"
+    echo "      --TARGET_DIR: The target directory to put box and log file"
+    echo "    Optional Arguments:"
+    echo "      --CACHE_IMAGE_DIR: The directory of cache images(ovf, vmdk)"
+}
+
+#########################################
+#
+#  check dependencies
+#
+#########################################
+checkDependencies(){
+    set +e
+    echo "[Info] Start to check the dependencies"
+    packer_path=$( which packer )
+    if [ -z "$packer_path" ];then
+        echo "[ERROR] packer is not installed . please install it."
+        exit 5
+    fi
+    vagrant_path=$( which vagrant )
+    if [ -z "$vagrant_path" ];then
+        echo "[ERROR] vagrant is not installed . please install it."
+        exit 5
+    fi
+    jq_path=$(which jq)
+    if [ ! -n "$(which jq)" ] ; then
+        echo "${ERROR_HEADER} jq is not installed . please install it, e.x. sudo apt-get install jq... Aborting. "
+        exit 5
+    fi
+    set -e
+}
+
+#########################################
+#
+#  prepare environment
+#
+#########################################
+prepareEnv(){
+    set +e
+    echo "[Info] Prepare environment"
+    PACKER=packer
+    if [ -x /opt/packer/packer ];then
+        PACKER=/opt/packer/packer
+    fi
+    PACKER_CACHE_DIR=$HOME/.packer_cache
+    # Enable Verbose Packer Logging
+    # see https://www.packer.io/docs/other/environmental-variables.html for details
+    export PACKER_LOG=1
+    export PACKER_LOG_PATH=./packer-debug.log
+    export PACKER_NO_COLOR=1  # for Jenkins usage. if manual run, suggest to turn color on (set to 0)
+    pkill $PACKER
+    bash $DIR/cleanup_vbox.sh
+    echo "[Info] Show current environment"
+    vagrant -v
+    $PACKER -v  # $? of "packer -v" is 1 ...
+    set -e
+}
+
+build(){
+    pushd $DIR
+    echo "[Info] Generate a parameter file pass to packer"
+    CFG_FILE=template.cfg
+    # parameter file pass to packer
+    echo {                                                        > $CFG_FILE
+    echo  \"playbook\": \"rackhd_ci_builds_mini\",                >> $CFG_FILE
```

Comment:
how abort ```rackhd_release_mini.yml```?

Reply:
The ```rackhd_ci_builds_mini.yml``` and ```rackhd_release_mini.yml``` both work.
Do you mean using ```rackhd_release_mini.yml``` here?